plugins/couchbase/CMakeLists.txt

```diff
@@ -50,10 +50,25 @@ IF ( COUCHBASEEMBED )
         ADD_DEFINITIONS(-D_USRDLL -DCOUCHBASEEMBED_EXPORTS)
         HPCC_ADD_LIBRARY(couchbaseembed SHARED ${SRCS})
         install(CODE "set(ENV{LD_LIBRARY_PATH} \"\$ENV{LD_LIBRARY_PATH}:${PROJECT_BINARY_DIR}:${PROJECT_BINARY_DIR}/libcouchbase-build/lib\")")
-        install(
-            TARGETS couchbaseembed
-            DESTINATION plugins
-            CALC_DEPS)
+        
+        if(WIN32 OR APPLE)
+            install(
+                TARGETS couchbaseembed
+                DESTINATION plugins
+                CALC_DEPS)
+        else()
+            install(
+                RUNTIME_DEPENDENCY_SET couchbaseembed_deps
+                DIRECTORIES "${VCPKG_FILES_DIR}/vcpkg_installed"
+                PRE_EXCLUDE_REGEXES "libcrypto|libssl"
+                POST_INCLUDE_REGEXES "^${VCPKG_FILES_DIR}\/vcpkg_installed\/.*"
+                POST_EXCLUDE_REGEXES ".*")
+            install(
+                TARGETS couchbaseembed
+                RUNTIME_DEPENDENCY_SET couchbaseembed_deps
+                DESTINATION plugins)
```

Comment:
CALC_DEPS was grabbing to much. It did a great job of including all the dependencies, unfortunately some of those dependencies are built into the platform by default. So this RUNTIME_DEPENDENCY_SET is a way to have more control over the process. Allows us to find dependencies of <plugin library> under vcpkg_installed, and the PRE_EXCLUDE_REGEXES will sort out any of the linked sonames so we don't even search for them, removing the chance of package conflicts.